nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -1110,18 +1110,7 @@ def dummy():
         # Return the output tensor of encoder and transpose from [seq_len, batch, hidden] to [batch, seq_len, hidden]
         return output_tensor.transpose(1, 0)
 
-    def decode(
-        self,
-        tokens_enc,
-        enc_mask,
-        num_tokens_to_generate,
-        encoder_input=None,
-        tokenizer=None,
-        enc_output=None,
-        enc_output_attn_mask=None,
-        ignore_ids=[],
-        bos_id=None,  # If bos=None, will use tokenizer.bos_id unless explicitly set to something else.
-    ):
+    def _init_for_inference(self, tokens_enc, enc_output):
         # Check whether the DDP is initialized. This is needed when running inference outside of training loop.
         if parallel_state.is_unitialized():
```

Comment:
Is this piece of code used in other places? If so perhaps useful to put it in a utility method?